src/targeting.js

```diff
@@ -181,6 +181,48 @@ export function newTargeting(auctionManager) {
     return [];
   };
 
+  /**
+   * Returns filtered ad server targeting for custom and allowed keys.
+   * @param {targetingArray} targeting
+   * @param {string[]} allowedKeys
+   * @return {targetingArray} filtered targeting
+   */
+  function getAllowedTargetingKeyValues(targeting, allowedKeys) {
+    logInfo(`Allowed default targeting keys [ ${allowedKeys.join(', ')}]`);
+    const defaultKeyring = Object.assign({}, CONSTANTS.TARGETING_KEYS, CONSTANTS.NATIVE_KEYS);
+    const defaultKeys = Object.keys(defaultKeyring);
+    targeting.map(adUnit => {
+      const adUnitCode = Object.keys(adUnit)[0];
+      const keyring = adUnit[adUnitCode];
+      const keys = keyring.filter(kvPair => {
+        const key = Object.keys(kvPair)[0];
+        // check if key is in default keys, if not, it's custom, we won't remove it.
+        const isCustom = defaultKeys.filter(defaultKey => key.indexOf(defaultKeyring[defaultKey]) === 0).length === 0;
+        // check if key explicitly allowed, if not, we'll remove it.
+        const found = isCustom || allowedKeys.find(allowedKey => {
+          const allowedKeyName = defaultKeyring[allowedKey];
+          // we're looking to see if the key exactly starts with one of our default keys.
+          // (which hopefully means it's not custom)
```

Comment:
You are correct. This change is only concerned with removing default targeting keys. If a publisher would like to stop generating or sending a custom key that they have defined, that is within their purview.

The reason that I opted against exact matching is because bid landscape concatenates the default key to the bidder, meaning "hb_pb" becomes "hb_pb_mybidder". I didn't want to pre-generate all possible keys and filter on all possible keys.

The best argument for exact matching to me, was that a publisher filtering out a default key would filter out custom keys with a matching prefix. to use your example if a publisher filtered out the default key "hb_pb" then their custom key "hb_pb_classification" would also be filtered out. This argument is why I included verbose logging, allowing the publisher to see that they can't filter "hb_pb" without also filtering "hb_pb_classification"